File: 백준/Silver/11399. ATM/ATM.py
import sys

N = int(sys.stdin.readline().rstrip())

atm_jul = list(map(int,sys.stdin.readline().split()))
new_atm = [ i for i in enumerate(atm_jul)]
new_atm.sort(key=lambda x : x[1])

res = 0
ans = []
for i in range(len(new_atm)):
    res += new_atm[i][1]
    ans.append(res)

print(sum(ans))

File: 백준/Silver/11399. ATM/ATM.java
import java.io.*;
import java.util.*;

class Main {
  static int N;
  static int answer = 0;
   public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

      N = Integer.parseInt(br.readLine());

      String[] st = br.readLine().split(" ");

      int[] waiting = new int[N];

      for (int i = 0; i < N; i++) {
        waiting[i] = Integer.parseInt(st[i]);
      }

      Arrays.sort(waiting);

      int sum = 0;

      for (int time : waiting) {
        sum += time;
        answer += sum;
      }

      System.out.println(answer);





   }
}